src/SourceBuild/tarball/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/DotNetHelper.cs

```diff
@@ -74,6 +76,21 @@ private static void InitNugetConfig()
 
             string nugetConfig = File.ReadAllText(nugetConfigPath);
             nugetConfig = nugetConfig.Replace("SMOKE_TEST_PACKAGE_FEED", Config.PrereqsPath);
+
+            // This package feed is optional.  You can use an additional feed of source-built packages to run the
+            // smoke-tests as offline as possible.
+            if (Config.CustomPackagesPath != null && Directory.Exists(Config.CustomPackagesPath))
+            {
+                nugetConfig = nugetConfig.Replace("CUSTOM_PACKAGE_FEED", Config.CustomPackagesPath);
+            }
+            else if (Config.CustomPackagesPath != null)
+            {
+                throw new ArgumentException($"Specified --with-packages {Config.CustomPackagesPath} does not exist.");
+            }
+            else
+            {
+                nugetConfig = string.Join(Environment.NewLine, nugetConfig.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries).Where(s => !s.Contains("CUSTOM_PACKAGE_FEED")).ToArray());
+            }
```

Comment:
Consider refactoring to avoid checking CustomPackagesPath mulitple times

```suggestion
            if (Config.CustomPackagesPath != null)
            {
                if (!Directory.Exists(Config.CustomPackagesPath))
                {
                    throw new ArgumentException($"Specified --with-packages {Config.CustomPackagesPath} does not exist.");
                }
                nugetConfig = nugetConfig.Replace("CUSTOM_PACKAGE_FEED", Config.CustomPackagesPath);
            }
            else
            {
                nugetConfig = string.Join(Environment.NewLine, nugetConfig.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries).Where(s => !s.Contains("CUSTOM_PACKAGE_FEED")).ToArray());
            }
```